sdks/universal-router-sdk/src/swapRouter.ts

```diff
@@ -43,6 +63,45 @@ export abstract class SwapRouter {
     })
   }
 
+  /**
+   * Builds the call parameters for a migration from a V3 position to a V4 position.
+   * Some requirements of the parameters:
+   *   - v3RemoveLiquidityOptions.collectOptions.recipient must equal v4PositionManager
+   *   - v3RemoveLiquidityOptions.liquidityPercentage must be 100%
+   *   - input pool and output pool must have the same tokens
+   *   - V3 NFT must be approved, or valid inputV3NFTPermit must be provided with UR as spender
+   */
+  public static migrateV3ToV4CallParameters(options: MigrateV3ToV4Options): MethodParameters {
+    const token0 = options.inputPosition.pool.token0
+    const token1 = options.inputPosition.pool.token1
+    invariant(token0 === options.outputPosition.pool.token0, 'TOKEN0_MISMATCH')
+    invariant(token1 === options.outputPosition.pool.token1, 'TOKEN1_MISMATCH')
+    invariant(options.v3RemoveLiquidityOptions.liquidityPercentage.equalTo(new Percent(100)), 'FULL_REMOVAL_REQUIRED')
```

Comment:
is your plan to make invariants to check the spender is UR and the recipient is v4positionmanager once we have those addresses?

Reply:
can we add this now? we should be checking like

spender == UR_ADDRESS[chainId] && recipient == UR_ADDRESS[chainId] and mock the chainId in the tests